File: app/src/main/java/com/developerxy/wildlegion/customviews/MySwitchPreference.java
package com.developerxy.wildlegion.customviews;

import android.content.Context;
import android.preference.PreferenceCategory;
import android.preference.SwitchPreference;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import com.developerxy.wildlegion.R;
import com.developerxy.wildlegion.utils.fonts.Typefaces;

public class MySwitchPreference extends SwitchPreference {

    private Context mContext;

    public MySwitchPreference(Context context) {
        super(context);
    }

    public MySwitchPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MySwitchPreference(Context context, AttributeSet attrs,
                              int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    protected void onBindView(View view) {
        super.onBindView(view);
        TextView title = view.findViewById(android.R.id.title);
        TextView summary = view.findViewById(android.R.id.summary);

        title.setTextColor(getContext().getResources().getColor(R.color.field_text_color));
        summary.setTextColor(getContext().getResources().getColor(R.color.field_text_color));

        title.setTypeface(Typefaces.Companion.getFont(getContext().getAssets(), "Bold"));
        summary.setTypeface(Typefaces.Companion.getFont(getContext().getAssets(), "Regular"));
    }
}

File: app/src/main/java/com/developerxy/wildlegion/customviews/MySwitchPreference.kt
package com.developerxy.wildlegion.customviews

import android.annotation.SuppressLint
import android.content.Context
import android.preference.SwitchPreference
import android.util.AttributeSet
import android.view.View


class MySwitchPreference : SwitchPreference, CustomFontPreference {

    @SuppressLint("NewApi")
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) :
            super(context, attrs, defStyleAttr, defStyleRes)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) :
            super(context, attrs, defStyleAttr)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context) : super(context)

    override fun onBindView(view: View) {
        super.onBindView(view)
        applyFont(context, view)
    }
}